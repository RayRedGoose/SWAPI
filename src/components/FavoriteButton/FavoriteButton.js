import React from 'react'
import PropTypes from 'prop-types'
import './FavoriteButton.scss'
import emptyStar from '../../images/star.svg'
import fullStar from '../../images/full-star.svg'

const FavoriteButton = (props) => {

  const star = (props.isFavorite === true) ? fullStar : emptyStar

  const toggleChange = () => {
    (!props.isFavorite)
      ? props.addFavorite(props.card)
      : props.removeFavorite(props.card.name)

    props.toggleCard()
  }

  return (
    <img
      src={star}
      alt="star"
      role="button"
      className="star-button"
      onClick={toggleChange}/>
  )
}

export default FavoriteButton

FavoriteButton.propTypes = {
  card: PropTypes.object.isRequired,
  isFavorite: PropTypes.bool.isRequired,
  toggleCard: PropTypes.func.isRequired,
  addFavorite: PropTypes.func,
  removeFavorite: PropTypes.func.isRequired
}
