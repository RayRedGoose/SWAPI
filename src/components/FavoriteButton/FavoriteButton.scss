.star-button {
  cursor: pointer;
  width: 40px;
}
