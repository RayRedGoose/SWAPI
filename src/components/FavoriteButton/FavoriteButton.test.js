import React from 'react'
import characterResults from '../../mockData/fakeCharacters.js'
import movieResults from '../../mockData/fakeMovies.js'
import { shallow } from 'enzyme'
import FavoriteButton from './FavoriteButton'

describe('FavoriteButton', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <FavoriteButton
        card={characterResults[1]}
        isFavorite={true}
        toggleCard={jest.fn()}
        addFavorite={jest.fn()}
        removeFavorite={jest.fn()}
      />);
  })

  it('should render FavoriteButton with all data passed in correctly', () => {
    expect(wrapper).toMatchSnapshot()
  })
})
