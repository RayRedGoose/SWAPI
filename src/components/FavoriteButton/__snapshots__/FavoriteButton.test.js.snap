// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FavoriteButton should render FavoriteButton with all data passed in correctly 1`] = `
<img
  alt="star"
  className="star-button"
  onClick={[Function]}
  role="button"
  src="full-star.svg"
/>
`;
